extensions/map-layers/src/mapLayers.ts

```diff
@@ -43,29 +45,33 @@ export class MapLayersUI {
     );
 
     MapLayersUI._iTwinConfig = config?.iTwinConfig;
+    MapLayersUI._featureInfoOpts = config?.featureInfoOpts;
+    MapLayersUI._mapLayerOptions = config?.mapLayerOptions;
 
-    MapLayersUI._uiItemsProviders.push(
-      new MapLayersUiItemsProvider({ ...config?.mapLayerOptions })
-    );
+    if (!config?.delayItemsProviderRegister)
+      MapLayersUI.registerUiItemsProviders();
+  }
+  public static registerUiItemsProviders(_mapLayerProviderOverrides?: UiItemProviderOverrides, _featureInfoProviderOverrides?: UiItemProviderOverrides) {
```

Comment:
nit: new line here plz

can this fn take an obj instead of a list of args? to future proof it, just in case there are other providers for this pkg later on